src-built-in/components/UIAPI/reducers/linker.ts

```diff
@@ -1,81 +1,46 @@
-                ...state,
-                processingRequest: true
-            };
-
-            const targetChannelName = toggleRequestState.channels[payload.channelID].name,
-                  targetChannelActive = toggleRequestState.channels[payload.channelID].active,
-                  targetWindowIdentifier = toggleRequestState.windowIdentifier;
-
-            const cmd = Cmd.run(linkChannel, {
-                successActionCreator: () => toggleSuccess(payload.channelID),
-                failActionCreator: () => toggleFailure(),
-                args: [targetChannelName, targetChannelActive, targetWindowIdentifier, toggleRequestState.actives]
-            });
-            return loop(toggleRequestState, cmd);
-
-        case ActionTypes.TOGGLE_CHANNEL_SUCCESS:
-            const toggleSuccessState = updateToggleChannelSuccessState(state, payload);
-            return toggleSuccessState;
-
-        case ActionTypes.TOGGLE_CHANNEL_FAILURE:
-            const toggleFailureState = {
-                ...state,
-                processingRequest: false
-            };
-            return toggleFailureState;
-
-        case ActionTypes.UPDATE_ACTIVE_CHANNELS:
-            const updateChannelsState = updateActiveChannelsState(state, payload);
-            return updateChannelsState;
-
-        case ActionTypes.UPDATE_ACTIVES:
-            const updateActivesState = {
-                ...state,
-                actives: payload.value
-            };
-            return updateActivesState;
-
-        case ActionTypes.LINKER_CLEANUP:
-            return loop(state, Cmd.run(cleanUp));
-
-        default:
-            return state;
-    }
-}
+const reducer = (state = initialState, action: ACTION_TYPES) => {
+    return produce(state, (draft: LinkerState) => {
+        actions.match(action, {
+            UPDATE_CHANNEL_STATUS: ({ channelId, active }) => {
+                draft.channels[channelId].active = active;
+            },
+            SET_CHANNELS: ({ channels }) => {
+                draft.channels = channels.map((channel: Channel, index: number) => {
+                    return {
+                        id: index,
+                        name: channel.name,
+                        color: channel.color,
+                        border: channel.border,
+                        active: false,
+                    }
+                });
+            },
+            UPDATE_ACTIVE_CHANNELS: ({ channelNames, windowIdentifier }) => {
+                const channelNameArray = Object.values(channelNames).map(obj => obj.name);
+                draft.windowIdentifier = windowIdentifier;
+                for (const channel of Object.values(draft.channels)) {
+                    channel.active = channelNameArray.includes(channel.name);
+                }
+            },
+            SET_ACCESSIBILITY: ({ isAccessibleLinker }) => {
+                draft.isAccessibleLinker = isAccessibleLinker;
+            },
+            default: a => draft,
```

Comment:
What is "a"? This should probably be renamed

Reply:
Has been! Good catch